client/src/components/choosers/page/PageChooserResult.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import $ from 'jquery';
+
+import { STRINGS } from '../../../config/wagtailConfig';
+
+const propTypes = {
+  isChoosable: PropTypes.bool.isRequired,
+  isNavigable: PropTypes.bool,
+  isParent: PropTypes.bool,
+  isFocused: PropTypes.bool,
+  onChoose: PropTypes.func.isRequired,
+  onNavigate: PropTypes.func.isRequired,
+  page: PropTypes.object.isRequired,
+  pageTypes: PropTypes.object,
+};
+
+const defaultProps = {
+  pageTypes: {},
+  isNavigable: false,
+  isParent: false,
+};
+
+// Capitalizes first letter without making any other letters lowercase
+function capitalizeFirstLetter(text) {
+  return text[0].toUpperCase() + text.slice(1);
+}
+
+class PageChooserResult extends React.Component {
+  renderTitle() {
+    const { isChoosable, onChoose, page } = this.props;
+
+    if (isChoosable) {
+      return (
+        <td className="title u-vertical-align-top" data-listing-page-title="">
+          <h2>
+            <a
+              onClick={onChoose}
+              className="choose-page"
+              href="#"
+              data-id={page.id}
+              data-title={page.title}
+              data-url={page.meta.html_url}
+              data-edit-url="/admin/pages/{page.id}/edit/"
+            >
+              {page.title}
+            </a>
+          </h2>
+        </td>
+      );
+    }
+
+    return (
+      <td className="title u-vertical-align-top" data-listing-page-title="">
+        <h2>
+          {page.title}
+        </h2>
+      </td>
+    );
+  }
+
+  renderUpdatedAt() {
+    const { page } = this.props;
+
+    if (page.meta.latest_revision_created_at) {
+      const updatedAt = moment(page.meta.latest_revision_created_at);
+
+      return (
+        <td className="updated u-vertical-align-top">
+          <div className="human-readable-date" title={updatedAt.format('D MMM YYYY h:mm a')}>
+            {updatedAt.fromNow()}
```

Comment:
It feels quite expensive to add `moment` as a dependency just for that. Is the intention to use it elsewhere? Otherwise I’d suggest either doing the formatting with vanilla JS, or using `date-fns` or `luxon`, or adding a formatted version of the date field(s) to the API response.

Reply:
I think, ideally, we would make use of Django's builtin date formats, not sure if we have a JS library for this yet?

Reply:
Actually, `fromNow` would be a tricky one to implement with that. I wonder if we could compile `moment` in such a way that it'll only fetch the locale it needs?

Reply:
It looks like there is now a way to do this in the standard library, as of ES2020: https://github.com/tc39/proposal-intl-relative-time.